ivadomed/loader/mri2d_segmentation_dataset.py

```diff
@@ -147,17 +147,28 @@ def prepare_indices(self):
                 if length > size:
                     raise RuntimeError('"length_2D" must be smaller or equal to image dimensions after resampling.')
 
-            for x in range(0, (shape[0] - self.length[0] + self.stride[0]), self.stride[0]):
-                if x + self.length[0] > shape[0]:
-                    x = (shape[0] - self.length[0])
-                for y in range(0, (shape[1] - self.length[1] + self.stride[1]), self.stride[1]):
-                    if y + self.length[1] > shape[1]:
-                        y = (shape[1] - self.length[1])
+            for x_min in range(0, (shape[0] - self.length[0] + self.stride[0]), self.stride[0]):
+                if x_min + self.length[0] > shape[0]:
+                    x_min = (shape[0] - self.length[0])
```

Comment:
I see this right align the mini patch to the entire shape, which might cause partial overlap with earlier patch. I assume this is an expect behaviour that would not affect training? similar for bottom aligning the mini patch for Y?

Reply:
Yes, this is normal and expected, we might optimize this behavior at a later time, but it does not affect the training significantly.